pkg/ride/tree_estimatorV4.go

```diff
@@ -42,6 +42,7 @@ func (e *treeEstimatorV4) estimate() (int, int, map[string]int, error) {
 	max := 0
 	m := make(map[string]int)
 	for i := 0; i < len(e.tree.Functions); i++ {
+		e.scope.resetFunctions()
```

Comment:
Why we should reset functions? I assume that every `FunctionDeclarationNode` creates new scope. Is it correct?

Reply:
Nope, during the estimation we use one scope for all callables, so it's dirtiness was not an issue then we don't check duplications. To check duplication correctly we have to clean the scope for every callable.